lib/async_hooks.js

```diff
@@ -339,10 +322,14 @@ function emitInitS(asyncId, type, triggerId, resource) {
 
 function emitBeforeN(asyncId) {
   processing_hook = true;
-  for (var i = 0; i < active_hooks_array.length; i++) {
-    if (typeof active_hooks_array[i][before_symbol] === 'function') {
-      runCallback(active_hooks_array[i][before_symbol], asyncId);
+  try {
+    for (var i = 0; i < active_hooks_array.length; i++) {
```

Comment:
The original reason for this was that the array index was used. This was before the design decision to not allow the callable hooks to be altered during execution of all hooks; not for benchmarking reasons. Though I'd say no sense in changing it unless benchmarks show there is an improvement.